### ISSUE TITLE
Setup IRC CTCP replies

### ISSUE THREAD

**Issue by zze0s** [autobrr/autobrr#1114]
Enable [CTCP replies](https://thelounge.chat/docs/guides/ctcp) with the IRC client.

**Comment by Kuredant** [comment]
Is there any concern other than setting `EnableCTCP` in `&ircevent.Connection{}`?

According to [irc-go/ircevent](https://github.com/ergochat/irc-go/blob/8d1f09a1b8d46e018ef369327057211860565e0e/ircevent/irc_ctcp.go#L48), this should automatically handle `VERSION`, `USERINFO`, `CLIENTINFO`, `TIME` and `PING`.

**Comment by KyleSanderson** [comment]
> Is there any concern other than setting `EnableCTCP` in `&ircevent.Connection{}`?
> 
> According to [irc-go/ircevent](https://github.com/ergochat/irc-go/blob/8d1f09a1b8d46e018ef369327057211860565e0e/ircevent/irc_ctcp.go#L48), this should automatically handle `VERSION`, `USERINFO`, `CLIENTINFO`, `TIME` and `PING`.

Yeah, and I'll document it here. We've been incredibly cautious about not making (visible) mistakes on networks and have limited mechanisms to actually send data. The only real place we attempt multiple authentication schemes is with NickServ, which too is limited. The last thing we want is for autobrr to be a weaponized vector on networks.

TIME and PING give relative server location which for our actual user I'm sure they'd prefer not to do.

If we have a bug, VERSION can be the equivalent of the nginx server string (which can contain more than the version, the version, or a confirmation that it's indeed autobrr). Which could aid in the weaponization of the product.

USERINFO and CLIENTINFO I also don't see it adding any value for our end users.

So, I'm not vetoing this or saying no, but without a good reason I see it as (likely) actively harmful to add. I heard that there was some request in passing from a help desk member, if any staff would like to have a discussion about our behaviour on networks we're (very) open to it.

**Comment by Kuredant** [comment]
Thanks for explaining your reasoning.
I don't have any argument in favor or against implementing CTCP replies, I was just wondering what the status of this issue was.
